### PR TITLE
Toggling proxy use

### PR DESCRIPTION
**Pull request by BardiaKh**
I tried to unify the model setting and server setting modals. Also added a switch for toggling proxy use one and off. 

image

As always, any feedback is appreciated.

### CODE REVIEW

File: src/components/settings/SettingsStore.tsx [hunk 1]
```diff
@@ -63,16 +64,15 @@ type ModelType = "openai" | "ollama";
 
 interface SettingsState {
 	showModelSettingsModal: boolean;
-	showServerSettingsModal: boolean;
 	openAIKey: string;
 	openAIBaseUrl: string;
 	serverProxyUrl: string;
 	ollamaUrl: string;
 	ollamaModel: string;
+	isLocal: boolean;
```
Comment (alishobeiri) [inline src/components/settings/SettingsStore.tsx:72]: Great change, thanks a lot 👍

**Comment by alishobeiri** [comment]
Thank you so much for the change, one small piece of feedback:
image

In the Ollama menu, there should not be a proxy option as everything should already be running locally, that means that if `ollama` is the model type, isLocal should be true by default. 
image

I would also make the toggle for `Use proxy` to instead be `Run API calls locally`. Ideally we can move away from the terminology of a proxy server as some folks found it confusing.

**Comment by BardiaKh** [comment]
@alishobeiri I think this is now ready for your review.

**Comment by alishobeiri** [comment]
Thanks a lot for the mention, will take a look now

**Comment by alishobeiri** [comment]
Just added some small changes as PR to your PR 😄 https://github.com/BardiaKh/thread/pull/2

**Comment by alishobeiri** [comment]
image
image

**Comment by BardiaKh** [comment]
Great!
Merged!

Will start adding the Anthropic support.

**Comment by alishobeiri** [comment]
Thanks a lot! Really appreciate it, I will make some changes to the modal selection modal to allow users to select from more than ollama and OpenAI

**Pull request merged**